### ISSUE TITLE
[amp-carousel] type=carousel does not update event index

### ISSUE THREAD

**Issue by shishirm** [ampproject/amphtml#15181]
## What's the issue?

The amp-carousel of type "slides" automatically updates the event.index when the slides are changed. However, the amp-carousel of type "carousel" does not update the event.index when a majority of a given slide is in view. This makes it difficult to use amp-bin and other bindable actions across different components. The slides type is not sufficient due to the look and feel of the component i.e. not being able to see all the possible slides and only being able to see the one slide.

## How do we reproduce the issue?

Your can reproduce this issue here: https://mshishir1.herokuapp.com/shop/amp_carousel_bug.html#development=1

## What browsers are affected?

All browsers and all device types should be affected.

## Which AMP version is affected?

This was likely always broken. I am currently reproducing this on AMP Version 1525892015019

**Comment by cathyxz** [comment]
This is an unimplemented feature. There is some ambiguity/disagreement around which "index" a type carousel `<amp-carousel>` is on during the scrolling process, which makes this a bit tricky.

**Comment by shishirm** [comment]
Is there some way to get more clarity around this and prioritize development? Maybe by how much of a given item in the carousel is in view for a given period of time before updating the index?

**Comment by ampprojectbot** [comment]
This issue hasn't been  updated in awhile. @cathyxz Do you have any updates?

**Comment by ampprojectbot** [comment]
This issue hasn't been  updated in awhile. @sparhami Do you have any updates?

**Comment by sparhami** [comment]
Closing as this appears to be a dupe of #14505

**Comment by nainar** [comment]
This issue is fixed for all users who upgrade their `<amp-carousel>` version to 0.2. 

For more details please see documentation here: https://github.com/ampproject/amphtml/blob/master/extensions/amp-carousel/amp-carousel.md
